docker/scripts/container_builder_base.sh

```diff
@@ -144,15 +144,25 @@ do
    TAG_OPTIONS="$TAG_OPTIONS -t $x"
 done
 
-run_docker_command "build $TAG_OPTIONS ." "$QUIET" || die "Failed to build the container image"
+# Look for prescense of docker buildx instance, otherwise create one
+# 'docker-container' is what the driver is called:
+# https://docs.docker.com/buildx/working-with-buildx/#build-multi-platform-images
+HAS_BUILD_X=$(docker buildx ls | grep 'docker-container' | cut -d ' ' -f 1)
```

Comment:
Theoretically, our customers can use these scripts to create their own images as well.  Does `buildx` come in `docker` by default or do they have to install additional dependencies?  If it is supported by default, how long has that support been there?  (I want to understand the minimum docker requirements for the customer)

Reply:
Yes buildx is bundled with docker desktop for windows, mac and Linux (rpm or deb) - https://docs.docker.com/buildx/working-with-buildx/

But it can be installed separately if a customer is missing it for some reason (using an older version of docker) - https://github.com/docker/buildx/

Looks like it's been bundled with docker desktop community edition since August 2019 - https://docs.docker.com/docker-for-windows/release-notes/#docker-desktop-community-2101

Reply:
Ok, it's been supported for almost two years.  It seems reasonable to rely on customers having it.  Ok, in general this looks good to me, but I'll defer to Dave on any script-fu related stuff.  Thanks Jonathan.